test/jdk/javax/swing/JButton/SwingButtonResizeTestWithOpenGL.java

```diff
@@ -0,0 +1,311 @@
+            System.out.println("Button loc: " + buttonLoc);
+            bimage = robot.createScreenCapture(new Rectangle(buttonLoc.x, buttonLoc.y, button.getWidth(), button.getHeight()));
+        } catch (Exception e) {
+            throw new RuntimeException("Problems capturing button image from " + "Robot", e);
+        }
+        return bimage;
+    }
+
+    //Save BufferedImage to PNG file
+    private void saveButtonImage(BufferedImage image, File file) {
+        if (image != null) {
+            try {
+                System.out.println("Saving button image to " + file.getAbsolutePath());
+                ImageIO.write(image, "PNG", file);
+            } catch (Exception e) {
+                throw new RuntimeException("Could not write image file");
+            }
+        } else {
+            throw new RuntimeException("BufferedImage was set to null");
+        }
+    }
+
+    private void disposeFrame() {
+        if (frame != null) {
+            frame.dispose();
+            frame = null;
+        }
+    }
+
+    private class DiffImage extends BufferedImage {
+
+        public boolean diff = false;
+        public int nDiff = -1;
+
+        Color bgColor;
+
+        int threshold = 0;
+
+        public DiffImage(int w, int h) {
+            super(w, h, BufferedImage.TYPE_INT_ARGB);
+            bgColor = Color.LIGHT_GRAY;
+        }
+
+        public int getNumDiffPixels() {
+            return nDiff;
+        }
+
+        public void compare(BufferedImage img1, BufferedImage img2) throws IOException {
+
+            int minx1 = img1.getMinX();
+            int minx2 = img2.getMinX();
+            int miny1 = img1.getMinY();
+            int miny2 = img2.getMinY();
+
+            int w1 = img1.getWidth();
+            int w2 = img2.getWidth();
+            int h1 = img1.getHeight();
+            int h2 = img2.getHeight();
+
+            if ((minx1 != minx2) || (miny1 != miny2) || (w1 != w2) || (h1 != h2)) {
+                // image sizes are different
+                throw new RuntimeException("img1: <" + minx1 + "," + miny1 + "," + w1 + "x" + h1 + ">" + " img2: " + minx2 + "," + miny2 + "," + w2 + "x" + h2 + ">" + " are different sizes");
+            }
+            System.out.println("Going to compare the images with a threshold of " + threshold + " pixels");
+            // Get the actual data behind the images
+            Raster ras1 = img1.getData();
+            Raster ras2 = img2.getData();
+
+            ColorModel cm1 = img1.getColorModel();
+            ColorModel cm2 = img2.getColorModel();
+
+            int r1;  // red
+            int r2;
+            int g1;  // green
+            int g2;
+            int b1;  // blue
+            int b2;
+
+            int x;
+            int y;
+
+            Object o1 = null;
+            Object o2 = null;
+            nDiff = 0;
+            for (x = minx1; x < (minx1 + w1); x++) {
+                for (y = miny1; y < (miny1 + h1); y++) {
```

Comment:
What's wrong with declaring the variable inside the for loop parameters? Condenses the code a bit too.